ioctl/cmd/ws/wsproject.go

```diff
@@ -0,0 +1,298 @@
+		config.Chinese: "w3bstream项目管理",
+	}
+
+	_flagProjectRegisterContractAddressUsages = map[config.Language]string{
+		config.English: "project register contract address",
+		config.Chinese: "项目注册合约地址",
+	}
+
+	wsProjectRegisterContractAddress string
+	wsProjectRegisterContractABI     abi.ABI
+
+	//go:embed wsproject.json
+	wsProjectRegisterContractJSONABI []byte
+	_                                = embed.FS{}
+)
+
+const (
+	createWsProjectFuncName  = "createProject"
+	createWsProjectEventName = "ProjectUpserted"
+	updateWsProjectFuncName  = "updateProject"
+	queryWsProjectFuncName   = "projects"
+)
+
+func init() {
+	var err error
+	wsProjectRegisterContractABI, err = abi.JSON(bytes.NewReader(wsProjectRegisterContractJSONABI))
+	if err != nil {
+		log.L().Panic("cannot get abi JSON data", zap.Error(err))
+	}
+
+	wsProject.AddCommand(wsProjectCreate)
+	wsProject.AddCommand(wsProjectUpdate)
+	wsProject.AddCommand(wsProjectQuery)
+
+	wsProject.PersistentFlags().StringVarP(
+		&wsProjectRegisterContractAddress,
+		"contract-address",
+		"c",
+		"",
+		config.TranslateInLang(_flagProjectRegisterContractAddressUsages, config.UILanguage),
+	)
+	_ = wsProject.MarkFlagRequired("contract-address")
+}
+
+func convertStringToAbiBytes32(hash string) (interface{}, error) {
+	t, _ := abi.NewType("bytes32", "", nil)
+
+	bytecode, err := hex.DecodeString(util.TrimHexPrefix(hash))
+	if err != nil {
+		return nil, err
+	}
+
+	if t.Size != len(bytecode) {
+		return nil, errors.New("invalid arg")
+	}
+
+	bytesType := reflect.ArrayOf(t.Size, reflect.TypeOf(uint8(0)))
+	bytesVal := reflect.New(bytesType).Elem()
+
+	for i, b := range bytecode {
+		bytesVal.Index(i).Set(reflect.ValueOf(b))
+	}
+
+	return bytesVal.Interface(), nil
+}
+
+func waitReceiptByActionHash(h string) (*iotexapi.GetReceiptByActionResponse, error) {
+	conn, err := util.ConnectToEndpoint(config.ReadConfig.SecureConnect && !config.Insecure)
+	if err != nil {
+		return nil, output.NewError(output.NetworkError, "failed to connect to endpoint", err)
+	}
+	defer conn.Close()
+	cli := iotexapi.NewAPIServiceClient(conn)
+	ctx := context.Background()
+
+	jwtMD, err := util.JwtAuth()
+	if err == nil {
+		ctx = metautils.NiceMD(jwtMD).ToOutgoing(ctx)
+	}
+
+	var rsp *iotexapi.GetReceiptByActionResponse
+	err = backoff.Retry(func() error {
+		rsp, err = cli.GetReceiptByAction(ctx, &iotexapi.GetReceiptByActionRequest{
+			ActionHash: h,
+		})
+		return err
+	}, backoff.WithMaxRetries(backoff.NewConstantBackOff(30*time.Second), 3))
+	if err != nil {
+		sta, ok := status.FromError(err)
+		if ok && sta.Code() == codes.NotFound {
+			return nil, output.NewError(output.APIError, "not found", nil)
+		} else if ok {
+			return nil, output.NewError(output.APIError, sta.Message(), nil)
+		}
+		return nil, output.NewError(output.NetworkError, "failed to invoke GetReceiptByAction api", err)
+	}
+	return rsp, nil
+}
+
+func getEventInputsByName(logs []*iotextypes.Log, eventName string) (map[string]any, error) {
```

Comment:
It's similar to `unpackEventParam` in blockindex/contractstaking/util.go, can we make a common util function?